cluster: add configs for read_replica to topic properties

LenaAn opened the pull request:
## Cover letter

Add read_replica and read_replica_bucket to topic_properties. Also add remote_topic_properties.
remote_topic_properties contain information that is fetched from remote
topic manifest: remote initial revision id and remote partition count.
This data should be stored in controller log in order to not download
manifest every time redpanda restarts.

Fixes: https://github.com/redpanda-data/redpanda/issues/4987

## Release notes
 - none

Lazin reviewed (commented):
That looks good, but will also need a property for an S3 bucket.

Lazin commented:
@LenaAn I don't think we need to allow updating read-replica property. The topic can't stop being read replica or switch to another bucket. In this case the user need to remove the topic and re-create it.

mmaslankaprv commented:
i think this is missing comapt testing between versions `-1` and `-2` for ADL. I am wondering if we even need to support read replicas in ADL serialized data structure ? Maybe it would be enough to just handle serde version and leave ADL version intact ?

LenaAn commented:
> i think this is missing comapt testing between versions `-1` and `-2` for ADL. I am wondering if we even need to support read replicas in ADL serialized data structure ? Maybe it would be enough to just handle serde version and leave ADL version intact ?

I'm fine with leaving ADL version as is, will update PR

LenaAn commented:
Added remote bucket, remote resivion id and remote partition count to the topic properties. Split original Pr to two, so the part with create_topic_request will be in the next PR.

Lazin reviewed (approved):
LGTM
